tasktiger/__init__.py

```diff
@@ -482,6 +485,108 @@ def get_queue_stats(self):
 
         return queue_stats
 
+    def purge_errored_tasks(
+        self,
+        queues=None,
+        exclude_queues=None,
+        last_execution_before=None,
+        limit=5000,
+    ):
+        """Purge failed tasks left in the ERROR state
+
+        Example usage::
+
+            limit = 500
+            n_processed = limit
+            while n_processed == limit:
+                n_processed = tiger.purge_errored_tasks(
+                    queues=['my-queue'],
+                    exclude_queues=['other-queue'],
+                    last_execution_before=(
+                        datetime.datetime.utcnow()
+                        - datetime.timedelta(days=14)
+                    ),
+                    limit=limit,
+                ):
+                time.sleep(.1)  # don't overload redis
+
+            # or to purge all in one go
+            tiger.purge_errored_tasks(limit=None)
+
+        :param iterable(str) queues: Queues to include
+        :param iterable(str) exclude_queues: Queues to exclude
+        :param datetime last_execution_before: If provided, only deletes tasks
+            older than this.
+        :param int limit: Stops after purging ``limit`` tasks. Can be
+            explicitly set to None to remove limit and purge all at once.
+
+        :returns: The total number of tasks purged
+        """
+        # some sanity checking and kind error messages for arguments
+        error_template = (
+            '{kwarg} should be an iterable of strings, not a string directly. '
+            'Did you mean `{kwarg}=[\'{val}\']`?'
+        )
+        assert not isinstance(queues, six.string_types), error_template.format(
+            kwarg='queues', val=queues
+        )
+        assert not isinstance(
+            exclude_queues, six.string_types
+        ), error_template.format(kwarg='exclude_queues', val=exclude_queues)
+        if last_execution_before:
+            assert isinstance(last_execution_before, datetime.datetime)
+
+        only_queues = set(queues or self.config['ONLY_QUEUES'] or [])
+        exclude_queues = set(
+            exclude_queues or self.config['EXCLUDE_QUEUES'] or []
+        )
+
+        def match(queue):
+            """
+            Returns whether the given queue should be included by checking each
+            part of the queue name.
+            """
+            for part in reversed_dotted_parts(queue):
+                if part in exclude_queues:
+                    return False
+                if part in only_queues:
+                    return True
+            return not only_queues
+
+        def errored_tasks():
+            queues_with_errors = self.connection.smembers(self._key(ERROR))
+            for queue in queues_with_errors:
+                if not match(queue):
+                    continue
+
+                skip = 0
+                total = None
+                task_limit = limit or 1000
```

Comment:
This seems a little weird setting a limit of `0` would end up being `1000`. Maybe just assert `limit > 0`?

Reply:
I've added an additional `assert` with the other argument validation